File: programmers_notes_core_scala/src/main/scala/org/sterzhen/programmers_notes/core/damain/InfoResource.scala
package org.sterzhen.programmers_notes.core.damain

/**
 * This entity encapsulates information about the resource
 *
 * @param id          resource id in DB
 * @param name        resource name
 * @param address     resource URL
 * @param description resource description
 */
class InfoResource(val id: Long, val name: String, val address: String, val description: String) {

    override def toString = s"InfoResource($id, $name, $address, $description)"
}

object InfoResource {
    def of(id: Long, name: String, address: String, description: String): InfoResource = {
        new InfoResource(id, name, address, description)
    }
}

File: programmers_notes_core/src/main/java/org/sterzhen/programmers_notes/core/domain/InfoResource.java
package org.sterzhen.programmers_notes.core.domain;

public class InfoResource {

    private Long id;
    private String name;
    private String address;
    private String description;

    public static InfoResource of(Long id, String name, String address, String description) {
        final InfoResource resource = new InfoResource();
        resource.setId(id);
        resource.setName(name);
        resource.setAddress(address);
        resource.setDescription(description);
        return resource;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "InfoResource{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", address='" + address + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}
